src/applications/disability-benefits/all-claims/utils/on-submit/purge-toxic-exposure-data.js

```diff
@@ -119,14 +126,19 @@ const purgeExposureDetails = (toxicExposure, exposureType, mapping) => {
   // Remove other/specify fields when:
   // - Field is null (orphaned data)
   // - User selected 'none' or has no selections
+  // - The corresponding 'other' selection is not true (e.g., otherExposures.other !== true)
   const hasOtherField = otherKey && otherKey in result;
   if (hasOtherField) {
     const isOtherFieldNull = result[otherKey] === null;
     const hasNoneSelected = result[exposureType]?.none;
     const hasNoSelections = !hasSelectedConditions(result[exposureType]);
+    const otherNotSelected = result[exposureType]?.other !== true;
```

Comment:
Am I misunderstanding that, because `other` will always be undefined, the "other" details field (e.g., `otherHerbicideLocations`, `specifyOtherExposures`) will be removed even if it has data.